cr_module/classes/redfish.py

```diff
@@ -317,40 +363,49 @@ def init_connection(self, reset=False):
         # if we have a connection object then just return
         if self.connection is not None:
             return
-
+        self.write_session_lock()
         self.get_credentials()
 
         # initialize connection
         try:
             self.connection = redfish.redfish_client(base_url=f"https://{self.cli_args.host}",
                                                      max_retry=self.cli_args.retries, timeout=self.cli_args.timeout)
         except redfish.rest.v1.ServerDownOrUnreachableError:
+            self.remove_session_lock()
```

Comment:
would suggest to move this statement to the `exit_on_error` method

Reply:
Done in second commit. This function can't be static because I need the self object in remove_session_lock(). Also I can't call this methode to print out that the session is locked because this will delete the lock. This is the only case where this is necessary so the exit without the function.